### PR TITLE
Feature/show task updater info

### PR DESCRIPTION
**Pull request by AnujChhikara**
Date: 13 Nov 2024

Developer Name: @AnujChhikara 

---

## Issue Ticket Number

  [Implement User Attribution for Task Progress Updates](https://github.com/Real-Dev-Squad/website-status/issues/1280) 
## Description
This PR  implementing a feature to display the details of the user who made the task progress update. The username will be shown as a clickable link that leads to their profile, ensuring accountability and transparency in task progress updates.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Updated UI</summary>

</details>

<details>
<summary>Dev Flag Working</summary>

</details>

## Test Coverage
<details>
<summary>
Screenshot 1</summary>

image
image

</details>

### CODE REVIEW

**Comment by pankajjs** [comment]
Design docs?

File: __tests__/Unit/Components/Tasks/ProgressUpdateCard/ProgressUpdateCardPresentation.test.tsx [hunk 1]
```diff
@@ -123,7 +129,76 @@ describe('ProgressUpdateCardPresentation Component', () => {
             onCardClick: mockedOnCardClick,
         };
     });
+
+    it('should rotate the angle icon when expanded', () => {
+        (useRouter as jest.Mock).mockReturnValue({
+            query: { dev: 'false' },
+        });
```
Comment (prakashchoudhary07) [inline __tests__/Unit/Components/Tasks/ProgressUpdateCard/ProgressUpdateCardPresentation.test.tsx:136]: Why is this being mocked here?

File: __tests__/Unit/Components/Tasks/ProgressUpdateCard/ProgressUpdateCardPresentation.test.tsx [hunk 2]
```diff
@@ -123,7 +129,76 @@ describe('ProgressUpdateCardPresentation Component', () => {
...
+        const props = { ...initialProps, isExpanded: true };
+        renderWithRouter(
```
Comment (prakashchoudhary07) [inline __tests__/Unit/Components/Tasks/ProgressUpdateCard/ProgressUpdateCardPresentation.test.tsx:138]: Does this compnent need router access??

File: src/app/services/usersApi.ts [hunk 3]
```diff
@@ -39,6 +43,10 @@ export const usersApi = api.injectEndpoints({
             query: ({ searchString }) => `/users?search=${searchString}`,
             providesTags: ['Users'],
         }),
+        getUserDetailsById: build.query<UserResponse, UsernameQueryArgs>({
+            query: ({ searchString }) => `/users/userId/${searchString}`,
```
Comment (prakashchoudhary07) [inline src/app/services/usersApi.ts:47]: Please don't use this route, The route naming is not proper and we might deprecate it real soon

File: src/components/taskDetails/ProgressUpdateCard/LatestProgressUpdateCardPresentation.tsx [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
 
 export default function LatestProgressUpdateCardPresentation({
+    username,
     dataToShowState,
     tooltipText,
     onMoreOrLessButtonClick,
     dateInAgoFormat,
 }: LatestProgressUpdateCardPresentationProps) {
+    const router = useRouter();
+    const isDevMode = router.query.dev === 'true';
```
Comment (prakashchoudhary07) [inline src/components/taskDetails/ProgressUpdateCard/LatestProgressUpdateCardPresentation.tsx:20]: Does component needs to be aware of the route?

File: __tests__/Unit/Components/Tasks/ProgressUpdateCard/ProgressUpdateCardPresentation.test.tsx (hunk 1 above)
Comment (AnujChhikara) [reply __tests__/Unit/Components/Tasks/ProgressUpdateCard/ProgressUpdateCardPresentation.test.tsx:136]: As these changes are currently behind a feature flag, we need to mock `useRouter` to set the dev query parameter to true for testing.

File: src/components/taskDetails/ProgressUpdateCard/LatestProgressUpdateCardPresentation.tsx (hunk 4 above)
Comment (AnujChhikara) [reply src/components/taskDetails/ProgressUpdateCard/LatestProgressUpdateCardPresentation.tsx:20]: These changes are currently behind  feature flag, so to retrieve the dev query parameter, we need access to the route. Once the feature flag is removed in the future, this setup can be eliminated

File: __tests__/Unit/Components/Tasks/ProgressUpdateCard/ProgressUpdateCardPresentation.test.tsx (hunk 2 above)
Comment (AnujChhikara) [reply __tests__/Unit/Components/Tasks/ProgressUpdateCard/ProgressUpdateCardPresentation.test.tsx:138]: changes are currently behind  feature flag, we need to use `useRouter` to get the query params to conditionally render changes

**Pull request closed without merging**